nifi-nar-bundles/nifi-jslt-bundle/nifi-jslt-processors/src/main/java/org/apache/nifi/processors/jslt/JSLTTransformJSON.java

```diff
@@ -277,4 +327,68 @@ private String readTransform(final PropertyValue propertyValue) {
             throw new UncheckedIOException("Read JSLT Transform failed", e);
         }
     }
-}
+
+
+    protected JsonNode getNextJsonNode(final TransformStrategy transformStrategy, final JsonParser jsonParser) throws IOException {
+
+        if (APPLY_TRANSFORM_TO_ENTIRE_FLOWFILE.equals(transformStrategy)) {
+            return null;
+        }
+        return getJsonNode(jsonParser);
+    }
+
+    private JsonNode getJsonNode(JsonParser jsonParser) throws IOException {
+        while (true) {
+            final JsonToken token = jsonParser.nextToken();
+            if (token == null) {
+                return null;
+            }
+
+            switch (token) {
+                case START_ARRAY:
+                case END_ARRAY:
+                case END_OBJECT:
+                    break;
+                case START_OBJECT:
+                    return jsonParser.readValueAsTree();
+                default:
+                    throw new IOException("Expected to get a JSON Object but got a token of type " + token.name());
+            }
+        }
+    }
+
+    /**
+     * Enumeration of supported Output Strategies
+     */
+    enum TransformStrategy implements DescribedValue {
+        APPLY_TRANSFORM_TO_ENTIRE_FLOWFILE("APPLY_TRANSFORM_TO_ENTIRE_FLOWFILE", "Entire FlowFile", "Entire FlowFile"),
+        APPLY_TRANSFORM_TO_EACH_OBJECT("APPLY_TRANSFORM_TO_EACH_OBJECT", "Each JSON Object", "Each JSON Object");
+
+        private final String value;
```

Comment:
The value string does not need to be repeated and the `name()` method can be used inside `getValue()`:
```suggestion
        APPLY_TRANSFORM_TO_ENTIRE_FLOWFILE("Entire FlowFile", "Entire FlowFile"),
        APPLY_TRANSFORM_TO_EACH_OBJECT("Each JSON Object", "Each JSON Object");
```